Prepare for Renovate with reusable workflows

ghostwriter opened the pull request:
|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Adding Renovate to Laminas, Mezzio repositories. https://gist.github.com/weierophinney/49ca8ae2b916d898dd3d4162b26ff41b

- Configure minimum supported PHP version
- Update composer dependencies
- Update to reusable `Continuous Integration` workflow
- Update to reusable `Automatic Releases` workflow

ghostwriter committed:
Configure minimum supported PHP version

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

ghostwriter committed:
Update to reusable automatic-releases workflow

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

Ocramius reviewed (changes requested):
Generated `composer.lock` seems to be invalid - needs to be re-generated (or rolled back to just hash update)

ghostwriter commented:
Unable to locate package `php8.1-sqlsrv`

Ocramius commented on .github/workflows/continuous-integration.yml:
```diff
@@ -4,71 +4,8 @@ on:
-        uses: laminas/laminas-continuous-integration-action@v1
-        with:
-          job: ${{ matrix.job }}
-    services:
```
These services should probably still run

Ocramius commented:
Looks like checks didn't run, perhaps due to YAML structure/parsing error?

ghostwriter commented:
@Ocramius apparently, the `services` keyword is not supported yet for reusable workflows. 

https://docs.github.com/en/actions/using-workflows/reusing-workflows#supported-keywords-for-jobs-that-call-a-reusable-workflow

Ocramius commented:
Ooof, guess we'd keep our phpunit tasks written down manually here, while excluding them from the laminas-ci integration perhaps?

ghostwriter committed:
Update .laminas-ci.json

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

ghostwriter committed:
Update composer.lock

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

ghostwriter commented:
Removed reusable Continuous Integration workflow for now.

Ocramius commented on .laminas-ci.json:
```diff
@@ -8,7 +8,12 @@
-    }
+    "exclude": [
+        {
+            "name": "PHPUnit on PHP 8.1 with latest dependencies"
```
This really shouldn't be excluded :|

Ocramius commented on .laminas-ci.json:
```diff
@@ -8,7 +8,12 @@
+            "name": "PHPUnit on PHP 8.1 with latest dependencies"
+        },
+        {
+            "name": "PHPUnit on PHP 8.1 with lowest dependencies"
```
This really shouldn't be excluded :|

ghostwriter replied:
This PR would fail otherwise, PHP 8.1 is not supported yet.

The `"ignore_php_platform_requirements": {"8.1": true}` syntax is not working. 

PHP 8.1 Tests are still generated https://github.com/laminas/laminas-db/actions/runs/1887918057

Ocramius replied:
We do though :thinking: 

https://github.com/laminas/laminas-db/blob/feeb1ffe83793d33305950a48c0eb9e540976ef6/composer.json#L28

ghostwriter replied:
I meant PHP 8.1 CI matrix: Unable to locate package `php8.1-sqlsrv`

Because it can't find the packing, it automatically fails early, before reaching the steps where it uses the `services` defined for the container.

Ocramius replied:
Yeah, I'm wondering if we can perhaps just skip sqlsrv tests for 8.1, rather than the whole 8.1 pipeline

ghostwriter replied:
I thought about that, but in `laminas-ci.json` the `extensions` we define for **all** PHP version, cause the script to fail on PHP 8.1 because one or more of the extensions doesn’t exist.

``` 
"extensions": [
        "pdo-mysql",
        "pdo-pgsql",
        "pdo-sqlite",
        "mysqli",
        "pgsql",
        "sqlite3",
        "sqlsrv"
    ],
```

if we remove this then we will lose those extensions on all PHP versions.

weierophinney replied:
Actually, we need to update the laminas-continuous-integration-action. MS released v5.10.0 of the sqlsrv extension, and it has PHP 8.1 support. I'll add that to the CI action, and then we should be able to test 8.1 here.

weierophinney committed:
qa: allow testing on PHP 8.1

Updates to laminas-continuous-integration-action should allow installation of sqlsrv extension now.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney commented:
I wonder why no actions triggered with my last push...

Ocramius commented:
@weierophinney github partial outage

Ocramius commented:
We need to do something like pushing + squashing an empty commit here

weierophinney commented:
Yes... But waiting for github status to go green (actions still has
degraded performance).

On Wed, Mar 23, 2022, 12:20 PM Marco Pivetta ***@***.***>
wrote:

> We need to do something like pushing + squashing an empty commit here
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/laminas/laminas-db/pull/239#issuecomment-1076599353>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAGKV7L6HDC726PV2POX43VBNHGTANCNFSM5O5NRITQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

ghostwriter committed:
ci

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

weierophinney commented:
@ghostwriter I'll figure out why the CI container isn't working, and get things sorted. Your work is done here; thanks for the patch!

ghostwriter committed:
Update composer.json

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

ghostwriter committed:
Update composer.lock

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

ghostwriter committed:
PHP80 Deprecated `pg_errormessage()` → `pg_last_error()`

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

weierophinney reviewed (approved):
Assuming tests pass after this run, :ship:

(Note: previous run passed, but the latest commit removes usage of a deprecated function.)

weierophinney commented:
@ghostwriter You overwrote a change I'd pushed to `.laminas-ci.json`:

```json
    "ignore_php_platform_requirements": {
        "8.1": true
    }
```

That will allow deps to install in 8.1, which should allow tests to pass.

ghostwriter committed:
Update .laminas-ci.json

Signed-off-by: Nathanael Esayeas <nathanael.esayeas@protonmail.com>

weierophinney commented:
Thanks, @ghostwriter !
